engine/engine.go

```diff
@@ -308,6 +308,12 @@ func (e *Engine) NewRangeQuery(ctx context.Context, q storage.Queryable, opts pr
 }
 
 func (e *Engine) NewRangeQueryFromPlan(ctx context.Context, q storage.Queryable, opts promql.QueryOpts, root logicalplan.Node, start, end time.Time, step time.Duration) (promql.Query, error) {
+	if opts == nil {
+		opts = promql.NewPrometheusQueryOpts(false, e.lookbackDelta)
+	}
+	if opts.LookbackDelta() <= 0 {
+		opts = promql.NewPrometheusQueryOpts(opts.EnablePerStepStats(), e.lookbackDelta)
+	}
 	qOpts := e.makeQueryOpts(ctx, start, end, step, opts)
```

Comment:
It might get convoluted because we need to propagate the opts passed into the method to the fallback engine from Prometheus. And our query opts are a different type.

Reply:
Ok!